### ISSUE TITLE
SwitchSettingsTile is stuck after enabling and disabling

### ISSUE THREAD

**Issue by omerhertz** [GAM3RG33K/flutter_settings_screens#98]
I am trying to use a SwitchSettingsTile with childrenIfEnabled. However, when I enable it, then disable it, I can't enable it again.
You can see a screenshot video here- [https://www.loom.com/share/bc7e1328ae8a473b8c5b882166d7b4e1](url)
BTW, in the console, for the above video, I get the "device-connected: true" print but it's still seems like disabled.

```
SettingsScreen(
  title: "App Settings",
  children: [
    SettingsGroup(
      title: 'Device Details',
      children: <Widget>[
        SwitchSettingsTile(
          settingKey: 'remote-device',
          title: 'Remote Mode',
          defaultValue: true,
          enabledLabel: 'Yes',
          disabledLabel: 'No',
          leading: const Icon(Icons.settings_remote_outlined),
          onChange: (value) {
            logInfo('remote-device: $value');
          },
        ),
        SwitchSettingsTile(
          settingKey: 'device-connected',
          title: 'Is Connected?',
          defaultValue: false,
          enabledLabel: 'Yes',
          disabledLabel: 'No',
          leading: const Icon(Icons.connect_without_contact_outlined),
          onChange: (value) {
            logInfo('device-connected: $value');
          },
          childrenIfEnabled: <Widget>[
            TextInputSettingsTile(
              settingKey: 'text-settings',
              title: 'Text Settings',
              borderColor: Colors.blueAccent,
              errorColor: Colors.deepOrangeAccent,
            ),
          ],
        ),
      ],
    ),
  ],
),
```
I'm using the latest version (0.3.3-null-safety+1)
If someone knows what the problem is, I'll be happy to get some help
Thanks

**Comment by nyxkn** [comment]
Should have been fixed with #93

**Comment by nyxkn** [comment]
Also this is probably the same issue as #90

**Comment by omerhertz** [comment]
@nyxkn I reran my app with the master branch and it is working correctly. So I guess this issue can be closed

**Comment by GAM3RG33K** [comment]
@nyxkn I have merged the PRs, can you please close this ticket if it iss resolved with merged PRs?

**Comment by nyxkn** [comment]
I think all three related issues can probably be closed: this one, #90, and #81.
Pretty sure it was all the same issue that was sorted with #93.
